test/excluded_prefixes_collector_test.go

```diff
@@ -0,0 +1,228 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package test contains k8s integration tests
+package test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/edwarnicke/exechelper"
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/networkservicemesh/integration-k8s-kind/k8s"
+)
+
+const (
+	prefixesFilePath    = "/var/lib/networkservicemesh/config/excluded_prefixes.yaml"
+	collectorNamespace  = "excluded-prefixes-collector"
+	excludedPrefixesEnv = "EXCLUDE_PREFIXES_K8S_EXCLUDED_PREFIXES"
+)
+
+type ExcludedPrefixesSuite struct {
+	suite.Suite
+	options       []*exechelper.Option
+	alpinePodName string
+}
+
+type prefixes struct {
+	Prefixes []string
+}
+
+var kubeAdmPrefixes = []string{
+	"10.244.0.0/16",
+	"10.96.0.0/16",
+}
+
+var userConfigPrefixes = []string{
+	"134.8.0.0/16",
+	"64.5.12.0/24",
+}
+
+func (et *ExcludedPrefixesSuite) SetupSuite() {
+	writer := logrus.StandardLogger().Writer()
+
+	et.options = []*exechelper.Option{
+		exechelper.WithStderr(writer),
+		exechelper.WithStdout(writer),
+	}
+
+	et.Require().NoError(exechelper.Run("kubectl apply -f ../deployments/prefixes-collector/collector-namespace.yaml", et.options...))
+	et.Require().NoError(exechelper.Run("kubectl apply -f ../deployments/prefixes-collector/collector-account.yaml", et.options...))
+	et.Require().NoError(exechelper.Run("kubectl apply -f ../deployments/prefixes-collector/collector-cluster-role.yaml", et.options...))
+	et.Require().NoError(exechelper.Run("kubectl apply -f ../deployments/prefixes-collector/config-map.yaml", et.options...))
+
+	et.setupAlpine()
+}
+
+func (et *ExcludedPrefixesSuite) TearDownTest() {
+	k8s.ShowLogs(collectorNamespace, et.options...)
+
+	et.Require().NoError(exechelper.Run("kubectl delete deployment -n excluded-prefixes-collector excluded-prefixes-collector", et.options...))
+	et.Require().NoError(exechelper.Run("kubectl delete pods -n excluded-prefixes-collector -l app=excluded-prefixes-collector --now", et.options...))
+	logrus.Info("Collector deleted")
+}
+
+func (et *ExcludedPrefixesSuite) TearDownSuite() {
+	k8s.ShowLogs(collectorNamespace, et.options...)
+
+	et.Require().NoError(exechelper.Run("kubectl delete -f ../deployments/prefixes-collector/collector-cluster-role.yaml", et.options...))
+	et.Require().NoError(exechelper.Run("kubectl delete -f ../deployments/prefixes-collector/collector-namespace.yaml --now", et.options...))
+}
+
+func (et *ExcludedPrefixesSuite) TestWithKubeAdmConfigPrefixes() {
+	et.Require().NoError(exechelper.Run("kubectl apply -f ../deployments/prefixes-collector/collector.yaml", et.options...))
+
+	et.Eventually(et.checkPrefixes(kubeAdmPrefixes), time.Second*120, time.Second*5)
```

Comment:
Why are we waiting for 120 seconds? Can we create a constant with a default timeout?

Reply:
Timeout is 120 sec, because total delay from the moment when the ConfigMap is updated to the moment when new keys are projected to the pod can be as long as kubelet sync period (1 minute by default) + ttl of ConfigMaps cache (1 minute by default) in kubelet ([source](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically)). Created const variables for default timeout and default tick.